shared/src/ledger/storage/merkle_tree.rs

```diff
@@ -1,44 +1,63 @@
 //! The merkle tree in the storage
-
-use std::convert::TryInto;
+use std::convert::{TryFrom, TryInto};
 use std::fmt;
+use std::marker::PhantomData;
 use std::str::FromStr;
 
+use arse_merkle_tree::default_store::DefaultStore;
+use arse_merkle_tree::error::Error as MtError;
+use arse_merkle_tree::traits::{Hasher, Value};
+use arse_merkle_tree::{
+    Hash as SmtHash, Key as TreeKey, SparseMerkleTree as ArseMerkleTree, H256,
+};
 use borsh::{BorshDeserialize, BorshSerialize};
 use ics23::commitment_proof::Proof as Ics23Proof;
 use ics23::{
     CommitmentProof, ExistenceProof, HashOp, LeafOp, LengthOp,
     NonExistenceProof, ProofSpec,
 };
+use itertools::Either;
```

Comment:
maybe switch to [this `either`](https://docs.rs/either/latest/either/)? it has some nice methods, such as [`map_left`](https://docs.rs/either/latest/either/enum.Either.html#method.map_left)

Reply:
it is a relatively small dep, with optional `serde` encoding which can be disabled so build times are smaller